File: app/src/main/java/com/sjn/taggingplayer/db/PlaylistSong.java
package com.sjn.taggingplayer.db;

import java.util.Date;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

@Accessors(prefix = "m")
@AllArgsConstructor(suppressConstructorProperties = true)
@NoArgsConstructor
@Getter
@Setter
public class PlaylistSong extends RealmObject {

    @PrimaryKey
    public long mId;
    public String mArtist;
    public String mTitle;
    public String mSongId;
    public Boolean mActive;
    public Date mCreatedAt;
    public Date mUpdatedAt;
    public Playlist mPlaylist;
}

File: app/src/main/java/com/sjn/stamp/db/PlaylistSong.kt
package com.sjn.stamp.db

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import java.util.*

open class PlaylistSong(
        @PrimaryKey var id: Long = 0,
        var artist: String? = null,
        var title: String? = null,
        var songId: String? = null,
        var active: Boolean? = null,
        var createdAt: Date? = null,
        var updatedAt: Date? = null,
        var playlist: Playlist? = null
) : RealmObject()